DataConnectors/CEF/cef_AMA_troubleshoot.py

```diff
@@ -783,36 +805,65 @@ def find_dcr_cloud_environment():
         return DEFAULT_MACHINE_ENV
 
 
-def getargs(should_print=True):
+def getargs():
     """
     Get execution args using argparse lib
     """
-    parser = argparse.ArgumentParser(description=SCRIPT_HELP_MESSAGE)
-    parser.add_argument('collect', nargs='?', help='Collect syslog message samples to file')
-    # parser.add_argument('--CEF', '--cef', action='store_true', default=False, help='Validate CEF DCR and events')
-    # parser.add_argument('--ASA', '--asa', action='store_true', default=False, help='Validate Cisco ASA DCR and events')
+    global STREAM_SCENARIO
+    parser = argparse.ArgumentParser()
+    parser.add_argument('collect', nargs='?',
+                        help='runs the script in collect mode. Useful in case you want to open a ticket.')
+    parser.add_argument('--CEF', '--cef', action='store_true', default=False,
+                        help='run the troubleshooting script for the CEF scenario.')
+    parser.add_argument('--ASA', '--asa', action='store_true', default=False,
+                        help='run the troubleshooting script for the Cisco ASA scenario.')
+    parser.add_argument('--FTD', '--ftd', action='store_true', default=False,
+                        help='run the troubleshooting script for the Cisco FTD scenario.')
     args = parser.parse_args()
+    if args.ASA:
+        STREAM_SCENARIO = "asa"
+    elif args.FTD:
+        STREAM_SCENARIO = "ftd"
+    else:
+        STREAM_SCENARIO = "cef"
+    return args
+
+
+def print_scenario(args, should_print=True):
```

Comment:
I think this is way too much logic, also should_print is not in use

suggestion:

def print_scenario(args):
    """
    param: args: the arguments returned from the getargs function
    """
    if list(vars(args).values()).count(True) > 1:
        print_error("More than 1 stream provided. Please run the script again with only one scenario.\n"
                    "For more information run 'python cef_AMA_troubleshoot.py -h'. Exiting.")
        sys.exit(1)
    else:
        print_notice("The scenario chosen is: {}".format(STREAM_SCENARIO.upper()))

Reply:
I removed the flag.
This logic won't work since in the case we get both a stream scenario and the collection flag, the script will fail.